fastqpreprocessing/src/utilities.cpp

```diff
@@ -0,0 +1,308 @@
+                break;
+            printf("option %s", long_options[option_index].name);
+            if (optarg)
+                printf(" with arg %s", optarg);
+            printf("\n");
+            break;
+        case 'b':
+            options.barcode_length = atoi(optarg);
+            break;
+        case 'u':
+            options.umi_length = atoi(optarg);
+            break;
+        case 'B':
+            options.bam_size = atof(optarg);
+            break;
+        case 's':
+            options.sample_id = string(optarg);
+            break;
+        case 'I':
+            options.I1s.push_back(string(optarg));
+            break;
+        case 'R':
+            options.R1s.push_back(string(optarg));
+            break;
+        case 'r':
+            options.R2s.push_back(string(optarg));
+            break;
+        case 'w':
+            options.white_list_file = string(optarg);
+            break;
+        case '?':
+        case 'h':
+          i = 0;
+          printf("Usage: %s [options] \n", argv[0]);
+          while (long_options[i].name != 0) {
+             printf("\t--%-20s  %-25s  %-35s\n", long_options[i].name, \
+                     long_options[i].has_arg == no_argument? \
+                     "no argument" : "required_argument", \
+                     help_messages[i]);
+             i = i + 1;
+          }
+          /* getopt_long already printed an error message. */
+          return;
+          break;
+        default:
+          abort();
+        }
+    }
+
+  // Check the options
+
+  // number of R1 and R2 files should be equal
+  if ((options.R1s.size() != options.R2s.size()) || (options.R1s.size() == 0)) {
+     std::cout << "ERROR: Unequal number of R1 and R2 fastq files in input\n";
+     std::cerr << "ERROR: Unequal number of R1 and R2 fastq files in input\n";
+     exit(1);
+  }
+
+  if ((options.I1s.size() != options.R1s.size()) and (options.I1s.size() != 0)) {
+     std::cout << "ERROR: Either the number of I1 input files are equal\n"
+                  "       to the number of R1 input files, or no I1 input files\n"
+                  "       should not be provided at all.\n";
+     std::cerr << "ERROR: Either the number of I1 input files are equal\n"
+                  "       to the number of R1 input files, or no I1 input files\n"
+                  "       should not be provided at all.\n";
+     exit(1);
+  }
+  // Bam file size must be positive
+  if (options.bam_size <= 0) {
+     std::cout << "ERROR: Size of a bam file (in GB) cannot be negative\n";
+     std::cerr << "ERROR: Size of a bam file (in GB) cannot be negative\n";
+     exit(1);
+  }
+
+  // must have a sample id
+  if (options.sample_id.size() == 0) {
+     std::cout << "ERROR: Must provide a sample id or name\n";
+     std::cerr << "ERROR: Must provide a sample id or name\n";
+     exit(1);
+  }
+
+  // barcode length must be positive
+  if (options.barcode_length <= 0) {
```

Comment:
Again, is 0 a valid option? Will that pipeline work with no barcodes or barcodes of length 1?

Reply:
barcode length cannot  be 0 or negative